calcom/cal.com
# fix: allow new booking to overlap old one when rescheduling

## What does this PR do?

When rescheduling we now also allow to book slots that were blocked from the original booking. 

Fixes #2668

Loom Video: https://www.loom.com/share/b782bcdb97894bf183ed526526e28240

## Type of change


## How should this be tested?

Test Steps to reproduce as described in linked issue: 

- Book a 60 minute booking using an event type which has Time-slot intervals set to 15 minutes. (For example at 11:30am)
- Reschedule the booking
- Select the same day as the original booking time
- See that now all slots from the original booking are available

Test with and without a connected calendar.

## Checklist
